include/deal.II/base/work_stream.h

```diff
@@ -1015,68 +1498,29 @@ namespace WorkStream
     Assert(chunk_size > 0, ExcMessage("The chunk_size must be at least one."));
     (void)chunk_size; // removes -Wunused-parameter warning in optimized mode
 
-    // if no work then skip. (only use operator!= for iterators since we may
+    // If no work then skip. (only use operator!= for iterators since we may
     // not have an equality comparison operator)
     if (!(begin != end))
       return;
 
-      // we want to use TBB if we have support and if it is not disabled at
-      // runtime:
-#  ifdef DEAL_II_WITH_TBB
-    if (MultithreadInfo::n_threads() == 1)
-#  endif
+    if (MultithreadInfo::n_threads() > 1)
       {
-        // need to copy the sample since it is marked const
-        ScratchData scratch_data = sample_scratch_data;
-        CopyData    copy_data    = sample_copy_data; // NOLINT
-
-        for (Iterator i = begin; i != end; ++i)
-          {
-            // need to check if the function is not the zero function. To
-            // check zero-ness, create a C++ function out of it and check that
-            if (static_cast<const std::function<
-                  void(const Iterator &, ScratchData &, CopyData &)> &>(worker))
-              worker(i, scratch_data, copy_data);
-            if (static_cast<const std::function<void(const CopyData &)> &>(
-                  copier))
-              copier(copy_data);
-          }
-      }
 #  ifdef DEAL_II_WITH_TBB
-    else // have TBB and use more than one thread
-      {
-        // Check that the copier exist
         if (static_cast<const std::function<void(const CopyData &)> &>(copier))
           {
-            // create the three stages of the pipeline
-            internal::Implementation2::
-              IteratorRangeToItemStream<Iterator, ScratchData, CopyData>
-                iterator_range_to_item_stream(begin,
-                                              end,
-                                              queue_length,
-                                              chunk_size,
-                                              sample_scratch_data,
-                                              sample_copy_data);
-
-            internal::Implementation2::Worker<Iterator, ScratchData, CopyData>
-              worker_filter(worker);
-            internal::Implementation2::Copier<Iterator, ScratchData, CopyData>
-              copier_filter(copier);
-
-            // now create a pipeline from these stages
-            tbb::pipeline assembly_line;
-            assembly_line.add_filter(iterator_range_to_item_stream);
-            assembly_line.add_filter(worker_filter);
-            assembly_line.add_filter(copier_filter);
-
-            // and run it
-            assembly_line.run(queue_length);
-
-            assembly_line.clear();
+            // If we have a copier, run the algorithm:
+            internal::tbb_no_coloring::run(begin,
```

Comment:
The patch is difficult to read due to whitespace changes here. Do I read this right that you continue to only hook in the TBB version, but not the taskflow version?